lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1808,7 +1808,7 @@ instance LiftHandler ErrCreateWallet where
 instance LiftHandler ErrWithRootKey where
     handler = \case
         ErrWithRootKeyNoRootKey wid ->
-            apiError err404 NoRootKey $ mconcat
+            apiError err410 NoRootKey $ mconcat
```

Comment:
I have updated this to `410` because there was some inconsistency. 
 - trying to send tx or join SP on HW wallet resutled in `410` and this error message
- updating password on HW wallet resulted in `404` and the same error message

Now all operations on HW wallet (requiring private key) result with `410`.
(On the other hand, I was thinking that if this code should be rather `401` or `403`, but maybe `410` fits as well...)

Reply:
I'd rather keep 401 for anything that is authentication related (if we end up having any form of authentication on the API, which is unlikely). 401 makes only sense in the presence of an `Authorization` header. 

403 would make sense here indeed. I haven't any strong preferences nevertheless 410 somewhat supposed that the root key was at some point available and removed later which doesn't really reflect what's happening here. I think the rationale was that in the past, the scenario were no root keys was found could only occur during the 3-step of the transaction creation: selection/signing/submission. On the second step, a the key is needed to sign a transaction and we do not have database atomicity between these steps, so _in theory_, the wallet could disappear between the "selection" and "signing" steps, resulting in an `NoRootKey` error, and therefore, leading to a 410 Gone. 

Now, returning 410 when updating passphrase doesn't really fit IMO, because the resource (root key) was never there in the first place. This was also the rationale behind 404. 

So, between:

a. Adding more logic to distinguish the cases where the key was there and gone because of a concurrent requests (410) and the case where it was never there to being with (404).

b. A more "generic" error code that fits both cases (403). 

I'd rather go for b. :man_shrugging:

Reply:
yep, option b seems reasonable (and rationale behind it)

Reply:
done in 399e928 - return 403 when operation on HW wallet requires private key